CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Models/SegoAponte2020/ViralInfectionVTMSteppables.py

```diff
@@ -0,0 +1,3222 @@
+        """
+        return [getattr(self, x.upper()) for x in self.bystander_targets]
+
+    @property
+    def bystander_target_data(self):
+        """
+        Bystander effect data
+
+        Keys are names of register cell types, items are probabilities of death
+        """
+        return self._bystander_targets.copy()
+
+
+class ChemotaxisSteppable(ViralInfectionVTMSteppableBasePy):
+    """
+    Implements immune cell chemotaxis module
+
+    By default, immune cells chemotaxis according to the module cytokine field.
+    The name of the field can be set with the attribute `target_field_name`.
+
+    The name of the immune cell type can be set with the attribute `immune_type_name`.
+
+    The chemotaxis parameter of immune cell types can be set with the attribute `lamda_chemotaxis`.
+    """
+
+    unique_key = ViralInfectionVTMLib.chemotaxis_steppable_key
+
+    def __init__(self, frequency=1):
+        super().__init__(frequency)
+
+        self._target_field_name = ''
+        self._lamda_chemotaxis = 0.0
+
+        # Initialize defaults
+        self.set_immune_type_name(immune_type_name)
+        self.set_target_field_name(cytokine_field_name)
+        self.set_lamda_chemotaxis(ViralInfectionVTMModelInputs.lamda_chemotaxis)
+
+    def start(self):
+        """
+        Called once to initialize simulation
+        """
+        for cell in self.cell_list_by_type(self.immune_type_id):
+            self.add_cell_chemotaxis_data(cell=cell)
+
+    def step(self, mcs):
+        """
+        Called every simulation step
+
+        :param mcs: current simulation step
+        :return: None
+        """
+        field = self.target_field
+        for cell in self.cell_list_by_type(self.immune_type_id):
+
+            cd = self.chemotaxisPlugin.getChemotaxisData(cell, self._target_field_name)
+            if cell.dict[ViralInfectionVTMLib.activated_cellg_key]:
+                cd.setLambda(self._lamda_chemotaxis / (1.0 + field[cell.xCOM, cell.yCOM, 1]))
+            else:
+                cd.setLambda(0)
+
+    def set_target_field_name(self, _name: str):
+        """
+        Set name of chemotaxis target field
+
+        :param _name: name of chemotaxis target field
+        :return: None
+        """
+        self._target_field_name = _name
+
+    @property
+    def target_field_name(self):
+        """
+        Name of chemotaxis target field
+        """
+        return self._target_field_name
+
+    @target_field_name.setter
+    def target_field_name(self, _name: str):
+        self.set_target_field_name(_name)
+
+    def set_lamda_chemotaxis(self, _val: float):
+        """
+        Set chemotaxis parameter value
+
+        :param _val: chemotaxis parameter value
+        :return: None
+        """
+        self._lamda_chemotaxis = _val
+
+    @property
+    def lamda_chemotaxis(self):
+        """
+        Chemotaxis parameter value
+        """
+        return self._lamda_chemotaxis
+
+    @lamda_chemotaxis.setter
```

Comment:
Still unsure what the purpose of defining a setter that calls a subsequent set_ function is?

Reply:
Simplified in https://github.com/covid-tissue-models/covid-tissue-response-models/pull/80/commits/a44ec678d95efc373d4328a9cfc0c6cbe3f432ad